src/engine/runtime.js

```diff
@@ -449,44 +542,124 @@ Runtime.prototype.stopAll = function () {
  * inactive threads after each iteration.
  */
 Runtime.prototype._step = function () {
+    if (this.pauseMode) {
+        // Don't do any execution while in pause mode.
+        return;
+    }
     // Find all edge-activated hats, and add them to threads to be evaluated.
     for (var hatType in this._hats) {
         var hat = this._hats[hatType];
         if (hat.edgeActivated) {
             this.startHats(hatType);
         }
     }
-    var inactiveThreads = this.sequencer.stepThreads(this.threads);
-    this._updateScriptGlows();
-    for (var i = 0; i < inactiveThreads.length; i++) {
-        this._removeThread(inactiveThreads[i]);
-    }
+    this.redrawRequested = false;
+    var inactiveThreads = this.sequencer.stepThreads();
+    this._updateGlows(inactiveThreads);
 };
 
+/**
+ * Set the current editing target known by the runtime.
+ * @param {!Target} editingTarget New editing target.
+ */
 Runtime.prototype.setEditingTarget = function (editingTarget) {
-    this._scriptGlowsPreviousFrame = [];
     this._editingTarget = editingTarget;
-    this._updateScriptGlows();
+    // Script glows must be cleared.
+    this._scriptGlowsPreviousFrame = [];
+    this._updateGlows();
 };
 
-Runtime.prototype._updateScriptGlows = function () {
+/**
+ * Set whether we are in pause mode.
+ * @param {boolean} pauseModeOn True iff in pause mode.
+ */
+Runtime.prototype.setPauseMode = function (pauseModeOn) {
+    // Inform the project clock/timer to pause/resume its time.
+    if (this.ioDevices.clock) {
+        if (pauseModeOn && !this.pauseMode) {
+            this.ioDevices.clock.pause();
+        }
+        if (!pauseModeOn && this.pauseMode) {
+            this.ioDevices.clock.resume();
+        }
+    }
+    this.pauseMode = pauseModeOn;
+};
+
+/**
+ * Set whether we are in 30 TPS compatibility mode.
+ * @param {boolean} compatibilityModeOn True iff in compatibility mode.
+ */
+Runtime.prototype.setCompatibilityMode = function (compatibilityModeOn) {
```

Comment:
Potential future change: what do you think about expressing this in terms of setting a target frame rate or frame time instead of a Boolean?